deploy/test-environments/fleet_api/src/agents_enrolled.py

```diff
@@ -36,23 +48,55 @@ def get_actual_agents() -> dict:
     return policies_dict
 
 
-def verify_agents_enrolled() -> bool:
+def verify_agent_count(expected: dict, actual: dict) -> bool:
     """
     Verify that the expected number of agents are enrolled
     """
-    expected = get_expected_agents()
-    actual = get_actual_agents()
     result = True
-    for policy_id, expected_count in expected.items():
+    for policy_id, expected_agents in expected.items():
         if policy_id not in actual:
             result = False
             logger.info(f"Policy {policy_id} not found in the actual agents mapping")
-        elif actual[policy_id] != expected_count:
+        elif actual[policy_id] != expected_agents.count:
             result = False
-            logger.info(f"Policy {policy_id} expected {expected_count} agents, but got {actual[policy_id]}")
+            logger.info(f"Policy {policy_id} expected {expected_agents.count} agents, but got {actual[policy_id]}")
```

Comment:
In the deployment run printout, there is currently no information available regarding the number of agents that are actively running. It could be beneficial to include this information for better context. In the case of a successful deployment, considering adding a log statement like
``` Python
else:
    logger.info(f"Policy id: {policy_id}, expected: {expected_agents.count}, actual: {actual[policy_id]}") would provide valuable insights.
```